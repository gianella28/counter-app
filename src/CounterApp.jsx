import { useState } from "react";
import  PropTypes, { object } from "prop-types";


export const CounterApp = ({ value }) => {
     
    const re=value;
    const [counter,setCounter] = useState(value);
    /*function handleAdd(event, newValue) {
        console.log(event);
        //console.log(newValue);
    }*/
    // 3 formas de hacer el handle
    
    const handleAdd = () => {
       // console.log(event);
        // hook forma 1
        //setCounter( counter + 1);
        // hook forma 2
        setCounter( (c) => c + 1)
    }

    const handleMenos = () => setCounter( counter - 1);
    
    const handleReset = () => {
        setCounter( value )
    }

    return (
        <>
        <h1>CounterApp</h1>
    
        <h2> { counter} </h2>
        <button onClick={  handleAdd }>+1</button>
        <button onClick={  handleMenos }>-1 </button>
        <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
 }