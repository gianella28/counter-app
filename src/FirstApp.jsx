// no hcer import { Fragment } from "react";
import PropTypes from 'prop-types';

const getuser=()=>({
  nombre:' giane'
});
const user=getuser();
/*const newMessage= {
  message:'hola mundo',
  title:' gia'
}  */ 
//funciones siempre poner fuera a menos ue sea necesario poner dentro del esport const first
const getResult = (a,b) =>{
  return a+b;
}
export const FirstApp = ( {title,subTitle,name} ) => {
 //console.log(props);

  return (
    <>
    {/*<h1>{JSON.stringify(user)}</h1>}
    {/*<code>{JSON.stringify( newMessage)}</code>
    <h1> {getResult(1,2)}</h1>*/}
    <h1 data-testid="test-title">{ title }</h1>

    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
   title: PropTypes.string.isRequired,
   subTitle: PropTypes.string,

}
FirstApp.defaultProps = {
  //title: 'no hay titulo',
  subTitle: 'no hay subtitulo',
  name:'gia'
}