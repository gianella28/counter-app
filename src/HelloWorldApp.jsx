/*export function App(){
    //document.createelement.....
   return (<h1>hola mundo 多多多多</h1> );
}

export const hola='123';

//export default app;*/


export const HelloWorldApp = () => {
  return (
    <div>Hello World App</div>
  ); 
}

//firstapp