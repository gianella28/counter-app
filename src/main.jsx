import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';
import './styles.css';
//renderizar la aplicacion
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
       {/* <CounterApp value={20} />
        <FirstApp  title="hola, soy vegeta"/>*/}
        <CounterApp value={10}/>
    </React.StrictMode>
);

