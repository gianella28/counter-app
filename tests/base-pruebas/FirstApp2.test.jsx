import { render, screen} from "@testing-library/react"
import { FirstApp } from "../../src/FirstApp"


describe('Pruebas en <firstapp/>',() => {
    const title ='hola soy gia';
    const subTitle= 'Soy un Subtitulo'
    test('debe de hacer watch con el snapshot',()=>{
        
        const {container} = render( <FirstApp title={ title } />);
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el mensaje "hola, soy gia"',()=>{
        render( <FirstApp title={ title }/>);
        expect (screen.getByText(title)).toBeTruthy();

    });

    test('debe de mostrar el titulo en un h1', () =>{
        render( <FirstApp title={ title }/>);
        expect (screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por los props',()=>{
        render( 
        <FirstApp 
            title={ title }
            subTitle={subTitle}
        />
        );
        expect( screen.getAllByText(subTitle).length).toBe(2);
    });
});