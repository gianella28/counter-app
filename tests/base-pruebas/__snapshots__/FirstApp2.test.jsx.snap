// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <firstapp/> debe de hacer watch con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy gia
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    gia
  </p>
</div>
`;
